### ISSUE TITLE
Contributed Libraries versions !?

### ISSUE THREAD

**Issue by helmarw** [fowlerk/ESP8266-Color-Weather-Station-v17#4]
can you specify the lib versions you were using also the version of the ESP8266 ?
is it 2.4.2 or later or already 2.5.x ?
lot have changed since a year ago.
im trying to downgrade everything as for October last year but still I cant get it compiled
also the versions of Adafruit_GFX and Adafruit_ILI9341 library seem to be crucial 

thnx
Helmar

**Comment by fowlerk** [comment]
@helmarw...give me a day or two and I will note the library and core versions I used when I compiled this last year.  Looking at the error messages, I assume you are probably correct in that a newer version of either the Adafruit GFX or ILI9341 libs has broken something.  This is not the first time that a change Adafruit has made to their display libs has caused an incompatibility with previously compiled code.

**Comment by helmarw** [comment]
Great! thnx a lot!
Best
Helmar

**Comment by fowlerk** [comment]
@helmarw...added current versions of contributed libs and core (including the IDE I use) to the README (I am using Windows, BTW).  Hope this helps.  I was able to recompile this version using the most up-to-date ESP8266 Weather Station code from ThingPulse.  At some point I will post a newer version of the code with much-improved error handling; just need a few more hours in the day ;).

**Comment by helmarw** [comment]
thnx!!! ill give it a try...

**Comment by helmarw** [comment]
![20191107_211642254_iOS](https://user-images.githubusercontent.com/10408121/68428698-f5ea5800-01ac-11ea-92e9-c155e3276919.jpg)

thnx, looks like its working :)))

**Comment by fowlerk** [comment]
@helmarw...looking good!  Though, it appears you are missing the large icon for the current weather conditions.  Did you clear all flash when uploading and go through the ESP data upload step to reload all the icons into flash?  Also make sure you set aside 3M for the SPIFFS file system, otherwise, all the icon files may not fit into flash.

**Comment by helmarw** [comment]
i don't know how the seen should look like, was happy that is displays anything ;)
yes i cleared the cache when uploading and i choose 3M as well
and when starting it the first time it downloaded quite while the icons i presume...
what could be the problem ?!

**Comment by helmarw** [comment]
now an icon just popped up "light rain"
i think it was broken clouds befor, maybe that one is missing ?

**Comment by fowlerk** [comment]
Did you run the ESP8266 Sketch Data Upload tool referenced in step 10 of the Quick Start in the README before the first run, after clearing SPIFFS?  This tool should now be an integral part of the IDE and doesn’t need to be separately downloaded as described in the Quick Start (at least under the Windows version).  This is what actually loads the correct icons.

If you run the code without this step, it has a fallback (part of the very original code from squix78 that I left in) that downloads icons from his website if they don’t exist in SPIFFS).  I probably should have removed this code, as the original icons no longer match up to all the weather conditions.  So, if you saw these being downloaded on first run, then you have an older and incomplete set of icons.

I’d recommend going back to the Data Upload tool and running it again and watching the output in the IDE monitor to make sure the current icon set (located in the /data directory) get loaded.  You shouldn’t have to clear or format SPIFFS manually as I believe the data upload tool now does this step before uploading the contents.

It looks like you are close, but you should always see a large icon to the left of the current temperature matching the weather conditions.

**Comment by helmarw** [comment]
oki, i ran the ESP Uploader, still raining here, ill see if things are changing.

on an other note, I've a couple of DHT22 left, but when im reading to your code im not sure
if its working or not, on some note is said no on other just uncomment to get it work
can you shed some light on it ?

**Comment by fowlerk** [comment]
The original code from squix78 (many generations back) contained support for a local temperature measurement using a DHT sensor.  When I forked his code several years ago, I did not have one and just commented out the code.  So, the original code is still there, but I have not tested it and frankly would be surprised if it worked without some rework of the code.  I am also not sure if this is supported any longer in ThingPulse’s current code.

**Comment by helmarw** [comment]
oki thnx

**Comment by helmarw** [comment]
added a few tweaks to get the DHT22 working
you can check the results at my fork of your code

**Comment by fowlerk** [comment]
Ah, nice work on this @helmarw, thanks for sharing.  If it’s okay with you, I’ll mention your fork in the README.

**Comment by helmarw** [comment]
yes please no problem, thnx :)

to get the DHT to work i also had to rewire the TFT slightly, but i made a shield for it too.
i ordered it yesterday at aisler.net, i can tell you next week if its working or not  ...

**Comment by helmarw** [comment]
the shield i made is working to of the box
https://github.com/helmarw/ESP8266-Color-Weather-Station-v17/blob/master/WeMOS_ILI9341_Shield/working_WeMos-ILI9341-Shield.png?raw=true

you can order it here if you need it: https://aisler.net/p/PJPTKIIB

**Comment by fowlerk** [comment]
Ah, very nice packaging @helmarw!  I’ve been tempted several times to make my own PCB’s for various projects, but have not done so yet.  Just the old fashioned way, with thru-hole components and lots of soldering on a breadboard ;).

**Comment by helmarw** [comment]
yes thats what i do a lot too, but since i discovered aisler, for small pcb its really worth it ordering,
3pcs for usually less than 20€
not even worth the time sitting in front of a cnc at work and milling it myself
